### PR TITLE
Add RunResources structure

### PR DESCRIPTION
**Pull request by mmsc2**
# TITLE

## Description

Description of the pull request changes and motivation.

## Checklist
- [x] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [x] This change requires new documentation.
  - [x] Documentation has been added/updated.
  - [x] CHANGELOG has been updated.

### CODE REVIEW

**Review by Oppen — APPROVED** [review]

File: src/utils.rs [hunk 1]
```diff
@@ -47,6 +47,35 @@ pub fn from_relocatable_to_indexes(relocatable: Relocatable) -> (usize, usize) {
...
     }
 }
+// ================
+//   RunResources
+// ================
+
+/// Maintains the resources of a cairo run. Can be used across multiple runners.
+#[allow(dead_code)]
+pub(crate) struct RunResources {
+    n_steps: Option<u64>,
```
Comment (fmoletta) [inline src/utils.rs:57]: Maybe we could make this a `usize` to keep it in line with `ExecutionResources.n_steps`

**Review by fmoletta — COMMENTED** [review]
Shouldn't this struct be used by the `CairoRunner` when running programs/contracts? Its just dead code if not.
Please fix clippy warnings

File: src/utils.rs (hunk 1 above)
Comment (mmsc2) [reply src/utils.rs:57]: noted, fixed

**Review by pefontana — CHANGES_REQUESTED** [review]
Same as in #1175 
I think that this PR https://github.com/lambdaclass/cairo-rs/pull/1181 will add this feature without making changes to the `CairoRunner.run_from_entrypoint` API
For now, let's leave this PR on stand by

**Pull request closed without merging**